pkg/api/steve/proxy/proxy.go

```diff
@@ -119,21 +119,7 @@ func (h *Handler) MatchNonLegacy(prefix string, force bool) gmux.MatcherFunc {
 		if !features.SteveProxy.Enabled() && !force {
 			return false
 		}
-
-		clusterID := strings.TrimPrefix(req.URL.Path, prefix)
-		clusterID = strings.SplitN(clusterID, "/", 2)[0]
-		if match.Vars == nil {
-			match.Vars = map[string]string{}
-		}
-		match.Vars["clusterID"] = clusterID
```

Comment:
This is mutating state, so it could be important.  It's setting "clusterID" which is scoped to the current request.  I'd probably leave it unless you can prove it's not needed.